corehq/apps/case_search/utils.py

```diff
@@ -188,3 +195,30 @@ def get_related_case_results(domain, cases, paths):
         results.extend(results_cache[path])
 
     return results
+
+
+def get_child_case_types(app, case_type):
+    """
+    Get child case types used by search detail tab nodesets in any modules
+    that match the given case type and are configured for case search.
+
+    Returns a set of case types
+    """
+    case_types = set()
+    for module in app.get_modules():
+        if module.case_type == case_type and module_offers_search(module):
+            for tab in module.search_detail("long").tabs:
+                if tab.has_nodeset and tab.nodeset_type == "subcase":
+                    case_types.add(tab.nodeset_identifier)
+
+    return case_types
+
+
+def get_child_case_results(domain, parent_cases, case_types):
+    parent_case_ids = {c.case_id for c in parent_cases}
+    child_cases = []
+    for case_type in case_types:
+        query = CaseSearchES().domain(domain).case_type(case_type).get_child_cases(parent_case_ids, "parent")
```

Comment:
I think you can pass the full list of case types here instead of doing repeated searches

Reply:
Cool, looks like it does work. 67b7678e449